profiling/src/lib.rs

```diff
@@ -950,6 +1072,18 @@ unsafe extern "C" fn alloc_profiling_malloc(len: u64) -> *mut ::libc::c_void {
         return ptr;
     }
 
+    let stats = ALLOCATION_PROFILING_STATS.with(|cell| {
+        let allocations = cell.borrow();
+        allocations.track_allocation(len)
+    });
+
+    if !stats.0 {
```

Comment:
What does this do? Access the first field? Why not access the field by name? This is difficult to read 🤔

Reply:
In a first version the `track_allocation()` function was inline, so the returned tuple and the usage were closer to each other, now with refactoring part of this, you are right. No one knows whats in the tuple anymore. I moved the rest of this (and also in the `realloc`-path to the `track_allocation()` function and the tuple is gone 🎉